dan_layer/engine/src/runtime/impl.rs

```diff
@@ -821,11 +830,13 @@ impl<TTemplateProvider: TemplateProvider<Template = LoadedTemplate>> RuntimeInte
         let substates_to_persist = self.tracker.take_substates_to_persist();
         self.invoke_modules_on_before_finalize(&substates_to_persist)?;
 
-        let (result, fee_receipt) = self.tracker.finalize(substates_to_persist)?;
+        let events = self.tracker.take_events();
```

Comment:
At this point we may need a new struct to define the tracker finalization, that will include `result`, `fee_receipt` and the `events` as fields.